lib/src/delegates/asset_picker_builder_delegate.dart

```diff
@@ -1404,30 +1424,39 @@ class DefaultAssetPickerBuilderDelegate
     BuildContext context,
     int index,
     List<AssetEntity> currentAssets, {
-    Widget? specialItem,
+    List<_SpecialItemModel> specialItemModels = const [],
   }) {
     final DefaultAssetPickerProvider p =
         context.read<DefaultAssetPickerProvider>();
     final int length = currentAssets.length;
     final PathWrapper<AssetPathEntity>? currentWrapper = p.currentPath;
     final AssetPathEntity? currentPathEntity = currentWrapper?.path;
 
-    if (specialItem != null) {
-      if ((index == 0 && specialItemPosition == SpecialItemPosition.prepend) ||
-          (index == length &&
-              specialItemPosition == SpecialItemPosition.append)) {
-        return specialItem;
+    final prependItems = <_SpecialItemModel>[];
+    final appendItems = <_SpecialItemModel>[];
+    for (final model in specialItemModels) {
+      switch (model.position) {
+        case SpecialItemPosition.prepend:
+          prependItems.add(model);
+        case SpecialItemPosition.append:
+          appendItems.add(model);
       }
     }
 
-    final int currentIndex;
-    if (specialItem != null &&
-        specialItemPosition == SpecialItemPosition.prepend) {
-      currentIndex = index - 1;
-    } else {
-      currentIndex = index;
+    if (prependItems.isNotEmpty) {
+      if (index < prependItems.length) {
+        return specialItemModels[index].item;
+      }
+    }
+
+    if (appendItems.isNotEmpty) {
+      if (index >= length + prependItems.length) {
+        return specialItemModels[index - length].item;
+      }
     }
 
+    final currentIndex = index - prependItems.length;
```

Comment:
If any of those items do not have a builder, this is likely to cause calculation issues. It's better to filter them first before accessing them in the code.

Reply:
So please add a null builder case in the example during the test to find out if there are any issues.

Reply:
done adding null builder case in example.